### PR TITLE
Changes done in initialize code path to set the hardware network interface  for PE node and memory servers

### PR DESCRIPTION
**Pull request by soumya-pn**
Changes done in initialize code path to set the hardware network interface  for PE node and memory servers

### CODE REVIEW

File: config/fam_pe_config.yaml [hunk 1]
```diff
@@ -19,6 +19,9 @@ runtime: PMIX
 # client_interface_type value of direct implies communication happens via function calls.
 client_interface_type: rpc
 
+# Device interface for communication for PE node
+#pe_if_device: ib0
```
Comment (mashoodk) [inline config/fam_pe_config.yaml:23]: I think you can just give if_device. Also add a comment that it is valid only for verbs provider

File: src/common/fam_ops_libfabric.h [hunk 2]
```diff
@@ -79,10 +79,10 @@ class Fam_Ops_Libfabric : public Fam_Ops {
      * @return - {true(0), false(1), errNo(<0)}
      */
 
-    Fam_Ops_Libfabric(bool is_source, const char *provider,
+    Fam_Ops_Libfabric(bool is_source, const char *provider, const char *if_device,
                       Fam_Thread_Model famTM, Fam_Allocator_Client *famAlloc,
                       Fam_Context_Model famCM = FAM_CONTEXT_DEFAULT);
-    Fam_Ops_Libfabric(bool source, const char *libfabricProvider,
+    Fam_Ops_Libfabric(bool source, const char *libfabricProvider, const char *if_device,
```
Comment (mashoodk) [inline src/common/fam_ops_libfabric.h:85]: git-clang-format

File: src/fam-api/fam_ops_libfabric.cpp [hunk 3]
```diff
@@ -63,13 +63,15 @@ Fam_Ops_Libfabric::~Fam_Ops_Libfabric() {
 }
 
 Fam_Ops_Libfabric::Fam_Ops_Libfabric(bool source, const char *libfabricProvider,
+		                     const char *pe_if_device,
```
Comment (mashoodk) [inline src/fam-api/fam_ops_libfabric.cpp:66]: I guess this code is common for memory server and pe. You can rename the variable to *if_device or if_device_str.

**Review by mashoodk — CHANGES_REQUESTED** [review]

**Review by mashoodk — APPROVED** [review]

File: src/common/fam_libfabric.cpp [hunk 4]
```diff
@@ -269,14 +269,20 @@ void libfabric_dump_profile_summary(void) {
  * @return - {true(0), false(1), errNo(<0)}
  */
 int fabric_initialize(const char *name, const char *service, bool source,
-                      char *provider, struct fi_info **fi,
+                      char *provider, char *if_device, struct fi_info **fi,
```
Comment (mashoodk) [inline src/common/fam_libfabric.cpp:272]: Can you add if_device argument description above.

**Review by mashoodk — CHANGES_REQUESTED** [review]

**Review by faizan-barmawer — CHANGES_REQUESTED** [review]
Changes look good.
Please rebase your git branch to the latest upstread/devel branch and do one minor change as suggested. 

Regards,
Faizan

**Review by faizan-barmawer — APPROVED** [review]
Code changes look good.

Regards,
Faizan

**Review by mashoodk — APPROVED** [review]

**Pull request merged**